actionpack/test/dispatch/content_security_policy_test.rb

```diff
@@ -302,6 +302,73 @@ def test_raises_runtime_error_when_unexpected_source
   end
 end
 
+class ContentSecurityPolicyMiddlewareTest < ActiveSupport::TestCase
+  def test_rack_lint
+    app = proc { [200, {}, []] }
+    env = Rack::MockRequest.env_for("", {})
+
+    assert_nothing_raised do
+      Rack::Lint.new(
+        ActionDispatch::ContentSecurityPolicy::Middleware.new(
+          Rack::Lint.new(app)
+        )
+      ).call(env)
+    end
+  end
+
+  def test_does_not_override_app_content_security_policy
+    app = proc { [200, { "content-security-policy" => "default-src 'self' https: http:" }, []] }
+    env = Rack::MockRequest.env_for("", {})
+
+    env["action_dispatch.content_security_policy"] = ActionDispatch::ContentSecurityPolicy.new do |p|
+      p.default_src -> { :self  }
+    end
+    env["action_dispatch.content_security_policy_nonce_generator"] = proc { "iyhD0Yc0W+c=" }
+    env["action_dispatch.content_security_policy_report_only"] = false
+    env["action_dispatch.show_exceptions"] = :none
+
+    _, headers, _ = Rack::Lint.new(
+      ActionDispatch::ContentSecurityPolicy::Middleware.new(Rack::Lint.new(app))
+    ).call(env)
+
+    assert_equal "default-src 'self' https: http:", headers["content-security-policy"]
+  end
+
+  def test_does_not_override_app_content_security_policy_for_rack_v2
+    app = proc { [200, { "Content-Security-Policy" => "default-src 'self' https: http:" }, []] }
+    env = Rack::MockRequest.env_for("", {})
+
+    env["action_dispatch.content_security_policy"] = ActionDispatch::ContentSecurityPolicy.new do |p|
+      p.default_src -> { :self  }
+    end
+    env["action_dispatch.content_security_policy_nonce_generator"] = proc { "iyhD0Yc0W+c=" }
+    env["action_dispatch.content_security_policy_report_only"] = false
+    env["action_dispatch.show_exceptions"] = :none
+
+    _, headers, _ = ActionDispatch::ContentSecurityPolicy::Middleware.new(app).call(env)
+
+    assert_equal "default-src 'self' https: http:", headers["content-security-policy"]
+  end
```

Comment:
This will fail when Rack v2 is no longer supported and `Rack::Headers` is not used in this middleware or https://github.com/rails/rails/blob/3a8b9baa51125567053869375147b69f31874c8f/actionpack/lib/action_dispatch/http/response.rb#L38-L40